src/js/select2/compat/matcher.js

```diff
@@ -8,6 +8,9 @@ define([
       if (params.term == null || $.trim(params.term) === '') {
         return match;
       }
+      else {
```

Comment:
the `params.term` is trimmed using a jQuery function so you could do

```js
params.term = $.trim(params.term)
if (params.term === '') {
  return match;
}
```

there is no need of an `else`

Reply:
@hakan0xFF But we have to be aware if `$.trim` works well when parameter is `null`. The result of `$.trim` is always a string, even blank? Or is there some case that it returns null? This should be included in the code.

Reply:
if the value/parameter is null the trim function returns an empty string. so it returns always a string. see https://github.com/select2/select2/blob/develop/vendor/jquery-2.1.0.js#L402